src/oatpp/core/data/mapping/type/Primitive.hpp

```diff
@@ -165,6 +173,25 @@ class String : public type::ObjectWrapper<std::string, __class::String> {
     return *this;
   }
 
+
+  inline bool equalsCI(const std::string &sb) {
+    const std::string& sa = this->m_ptr.operator*();
+    return (sa.size() == sb.size()) && std::equal(sa.begin(), sa.end(), sb.begin(),
+                      [] (char a, char b) -> bool {
+                          return std::tolower(a) == std::tolower(b);
+                      });
+  }
+
+  inline bool equalsCI(const String &b) {
+    const std::string& sb = *b;
+    return this->equalsCI(sb);
+  }
+
+  inline bool equalsCI(const char *b) {
```

Comment:
Should return `true` if both `this->m_ptr == nullptr` and `b == nullptr`
Should return `false` if `this->m_ptr != nullptr` and `b == nullptr`
Should return `false` if `this->m_ptr == nullptr` and `b != nullptr`

Reply:
Should be fine now.